### PR TITLE
[APM] Add obfuscation for OpenSearch traces

### PR DESCRIPTION
**Pull request by ichinaski**
### What does this PR do?

This PR adds obfuscation support for OpenSearch statements within span metadata.

OpenSearch spans have a similar structure to ElasticSearch ones, and the main difference lies in the `opensearch.*` vs `elasticsearch.*` tags. For example, `opensearch.body` vs `elasticsearch.body`.

While the tag keys differ, the span type can be either `elasticsearch` (dd-trace-js) or `opensearch` (dd-trace-java). 

The configuration is done in the same way as elasticsearch one, via environment variables (see releasenotes).

### Motivation

Jira ticket: https://datadoghq.atlassian.net/browse/APMSP-914
Support ticket: https://datadoghq.atlassian.net/browse/FRAPMS-3979

### CODE REVIEW

File: pkg/trace/config/config.go [hunk 1]
```diff
@@ -85,6 +85,9 @@ type ObfuscationConfig struct {
 	// ES holds the obfuscation configuration for ElasticSearch bodies.
 	ES obfuscate.JSONConfig `mapstructure:"elasticsearch"`
 
+	// OpenSearch holds the obfuscation configuration for OpenSearch bodies.
+	OpenSearch obfuscate.JSONConfig `mapstructure:"opensearch"`
```
Comment (ajgajg1134) [inline pkg/trace/config/config.go:89]: I think we also need some changes in `comp/trace/config/setup.go` to setup the configuration based on the incoming yaml (see the example on line 411 from elasticsearch)

Comment (ichinaski) [reply pkg/trace/config/config.go:89]: Thanks for the pointer. I updated the setup based on the config.

**Review by ajgajg1134 — COMMENTED** [review]
Overall looks good to me! I think it just needs a release-note so folks know that we're adding this and it's enabled by default

**Review by misteriaud — APPROVED** [review]
LGTM for code owned by ASC !

**Review by knusbaum — COMMENTED** [review]
Do we need to add this to the config template like the elasticsearch?
https://github.com/DataDog/datadog-agent/blob/30fa0a20418e8f6e3c1cacb9ca78441442361709/pkg/config/config_template.yaml#L1127-L1138

File: releasenotes/notes/add-opensearch-obfuscation-a26c678785bce376.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    APM: Add obfuscation support for opensearch statements within span metadata. This feature works in the same way as the existing elasticsearch one, and it's enabled by default. It is configured by binding ``apm_config.obfuscation.opensearch.*`` parameters to new obfuscation environment variables. In particular, bind:
```
Comment (rtrieu) [inline releasenotes/notes/add-opensearch-obfuscation-a26c678785bce376.yaml:11]:
```suggestion
    APM: Add obfuscation support for OpenSearch statements within span metadata. This feature works in the same way as the existing Elasticsearch one, and is enabled by default. It is configured by binding ``apm_config.obfuscation.opensearch.*`` parameters to new obfuscation environment variables. In particular, bind:
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Minor feedback but otherwise looks good.

**Review by rtrieu — APPROVED** [review]
LGTM

**Comment by ichinaski** [comment]
> Do we need to add this to the config template like the elasticsearch?
> 
> https://github.com/DataDog/datadog-agent/blob/30fa0a20418e8f6e3c1cacb9ca78441442361709/pkg/config/config_template.yaml#L1127-L1138

Good point! I added this in a new commit.

**Review by knusbaum — APPROVED** [review]
LGTM

**Pull request merged**